transmart-core-db/grails-app/services/org/transmartproject/db/support/InQuery.groovy

```diff
@@ -0,0 +1,72 @@
+package org.transmartproject.db.support
+
+import grails.orm.HibernateCriteriaBuilder
+import org.grails.datastore.mapping.query.api.Criteria
+import org.hibernate.Criteria as HibernateCriteria
+import org.hibernate.criterion.Restrictions
+import org.transmartproject.core.exceptions.InvalidRequestException
+
+/**
+ * This class aimed to overcome oracle limitation on the number of items in IN clause:
+ * Message: ORA-01795: maximum number of expressions in a list is 1000
+ *
+ * This class constructs composite condition
+ * Instead of making one condition
+ * patient_num IN (1, 2, 3, ... 1200)
+ * it chops list of items and join sub-conditions with OR operations
+ * patient_num IN (1, 2, 3, ... 1000) OR patient_num IN (1001, 1002, 1003, ... 1200)
+ */
+class InQuery {
+
+    public static final int MAX_LIST_SIZE = 1000
```

Comment:
This 1000 items constraint is only for oracle, maybe this code can generate the normal sql for postgres and only chop for oracle?